script/clear_external_cache.pl

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env perl
+use strict;
+use warnings;
+
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+use MusicBrainz::Server::Context;
+
+# This script can be used to clear everything in memcached. It assumes that
+# the CACHE_MANAGER_OPTIONS DBDefs setting resembles the default one (so it
+# has an "external" profile pointing to Cache::Memcached::Fast, and
+# default_profile is set to "external").
+
+my $c = MusicBrainz::Server::Context->create_script_context;
+$c->cache->flush_all;
```

Comment:
[Yes, it does.](https://github.com/metabrainz/musicbrainz-server/pull/215#discussion_r49938057) The problem is that memcached doesn’t support this ([docs](https://code.google.com/p/memcached/wiki/NewProgramming#Invalidating_by_Tag)). There is an undocumented [`stats cachedump`](http://blog.elijaa.org/index.php?post/2010/12/24/Understanding-Memcached-stats-cachedump-command) that will list items, but only to a hardcoded limit, so it‘s unreliable.

If we want to selectively remove items from cache, I think a better approach would be to retrieve those entities that have a recent `last_updated` timestamp from Postgres, then remove them from memcached.